MapboxNavigation/InstructionsBannerView.swift

```diff
@@ -80,16 +81,39 @@ open class BaseInstructionsBannerView: UIControl {
             baselineAlignInstructions()
         }
         
-        primaryLabel.instruction = primaryInstruction
+        primaryLabel.instruction = instruction.primaryTextComponents
         secondaryLabel.instruction = secondaryInstruction
+        maneuverView.visualInstruction = instruction
     }
     
     override open func prepareForInterfaceBuilder() {
         super.prepareForInterfaceBuilder()
         maneuverView.isStart = true
         
-        primaryLabel.instruction = [VisualInstructionComponent(type: .destination, text: "Primary text label", imageURL: nil)]
+        primaryLabel.instruction = [VisualInstructionComponent(type: .destination, text: "Primary text label", imageURL: nil, maneuverType: .none, maneuverDirection: .none)]
         
         distance = 100
     }
+    
+    /**
+     Updates the instructions banner for a given `RouteProgress`.
+     */
+    public func update(for currentLegProgress: RouteLegProgress) {
```

Comment:
Since this method only cares about `currentLegProgress.currentStepProgress`, it would be better if this method took a `RouteStepProgress` argument directly.

Dealer's choice though, I will be tackling this (and things like this) directly when I resurrect #993.